contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1184,6 +1036,128 @@ contract NodeOperatorsRegistry is AragonApp, Versioned {
+    /// @param _publicKey The public key of the validator.
+    /// @param _proofSlotTimestamp The timestamp (slot time) when the validator was last known to be in an active ongoing state.
+    /// @param _eligibleToExitInSec The number of seconds the validator was eligible to exit but did not.
+    /// @return True if the validator has exceeded the exit deadline threshold and hasn't been reported yet.
+    function isValidatorExitDelayPenaltyApplicable(
+        uint256, // _nodeOperatorId
+        uint256 _proofSlotTimestamp,
+        bytes _publicKey,
+        uint256 _eligibleToExitInSec
+    ) external view returns (bool) {
+        // Check if the key is already reported
+        if (isValidatorExitingKeyReported(_publicKey)) {
+            return false;
+        }
+        return _eligibleToExitInSec >= _exitDeadlineThreshold()
+            && _proofSlotTimestamp - _eligibleToExitInSec >= exitPenaltyCutoffTimestamp();
+    }
+
+    /// @notice Handles tracking and penalization logic for a node operator who failed to exit their validator within the defined exit window.
+    /// @dev This function is called by the StakingRouter to report the current exit-related status of a validator
+    ///      belonging to a specific node operator. It marks the validator as processed to avoid duplicate reports.
+    /// @param _nodeOperatorId The ID of the node operator whose validator's status is being delivered.
+    /// @param _proofSlotTimestamp The timestamp (slot time) when the validator was last known to be in an active ongoing state.
+    /// @param _publicKey The public key of the validator being reported.
+    /// @param _eligibleToExitInSec The duration (in seconds) indicating how long the validator has been eligible to exit after request but has not exited.
+    function reportValidatorExitDelay(
+        uint256 _nodeOperatorId,
+        uint256 _proofSlotTimestamp,
+        bytes _publicKey,
+        uint256 _eligibleToExitInSec
+    ) external {
+        _auth(STAKING_ROUTER_ROLE);
+        require(_publicKey.length == 48, "INVALID_PUBLIC_KEY");
+
+        // Check if exit delay exceeds the threshold
+        require(_eligibleToExitInSec >= _exitDeadlineThreshold(), "EXIT_DELAY_BELOW_THRESHOLD");
+        // Check if the proof slot timestamp is within the allowed reporting window
+        require(_proofSlotTimestamp - _eligibleToExitInSec >= exitPenaltyCutoffTimestamp(), "TOO_LATE_FOR_EXIT_DELAY_REPORT");
+
+        bytes32 processedKeyHash = keccak256(_publicKey);
+        // Skip if key is already processed (i.e., not in NotProcessed state)
+        if (_validatorProcessedLateKeys[processedKeyHash]) {
+            return;
+        }
+        // Mark the validator exit key as processed
+        _validatorProcessedLateKeys[processedKeyHash] = true;
+
+        emit ValidatorExitStatusUpdated(_nodeOperatorId, _publicKey, _eligibleToExitInSec, _proofSlotTimestamp);
```

Comment:
The event name ValidatorExitStatusUpdated doesn’t clearly convey that the validator was late to exit